### ISSUE TITLE
Is is possible to control the log level from the Java API ?

### ISSUE THREAD

**Issue by hervegirod** [alphacep/vosk-api#697]
I have a log of VLOGs (level 6) mainly at the end of the decoding phase. Is it possible to control the log level through the java API (or through a configuration file) ?

**Comment by hervegirod** [comment]
If I use the setLogLevel(LogLevel) method in Java, it seems to not do anything.

**Comment by nshmyrev** [comment]
> If I use the setLogLevel(LogLevel) method in Java, it seems to not do anything.

How do you use it exactly?

**Comment by hervegirod** [comment]
I am doing at the beginning after registering the library:
`LibVosk.setLogLevel(LogLevel.SILENT)`

To be fair, I have forked the Java code to add a `SILENT `enum state because I always had warnings when I ended the recognizer:
```
LibVosk.end(model);
recognizer.close();
```

The new code for the enum is:
```
public enum LogLevel {
   /**
    * Print nothing.
    */
   SILENT(-10), // silent
   /**
    * Print assert failed.
    */
   ASSERT_FAILED(-3), // assert failed
   /**
    * Print errors.
    */
   ERRORS(-2), // Print errors
   /**
    * Print warning and errors.
    */
   WARNINGS(-1), // Print warning and errors
   /**
    * Print info, along with warning and error messages, but no debug.
    */
   INFO(0), // Print info, along with warning and error messages, but no debug
   /**
    * Print debug info.
    */
   DEBUG(1);      // Print debug info

   private final int value;

   LogLevel(int value) {
      this.value = value;
   }

   public int getValue() {
      return this.value;
   }
}
```

I looked at the C code for Kaldi and it seemed that the value -10 was for not showing any log.

**Comment by hervegirod** [comment]
I have these kinds of logs at the end (everything is OK, with no logs, before I close):
```
VLOG[6] (VoskAPI:GetDeltaWeights():online-ivector-feature.cc:706) Weight for frame 49 changing from 0 to 0.001
VLOG[6] (VoskAPI:GetDeltaWeights():online-ivector-feature.cc:706) Weight for frame 50 changing from 0 to 0.001
VLOG[6] (VoskAPI:GetDeltaWeights():online-ivector-feature.cc:706) Weight for frame 51 changing from 0 to 0.001
VLOG[6] (VoskAPI:GetDeltaWeights():online-ivector-feature.cc:706) Weight for frame 52 changing from 0 to 0.001
VLOG[6] (VoskAPI:GetDeltaWeights():online-ivector-feature.cc:706) Weight for frame 53 changing from 0 to 0.001
VLOG[6] (VoskAPI:GetCutoff():lattice-faster-decoder.cc:677) Number of tokens active on frame 41 is 1558
VLOG[6] (VoskAPI:ProcessEmitting():lattice-faster-decoder.cc:729) Adaptive beam on frame 41 is 13
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:472) In linear CG: initial norm-square of residual = 0.00777745
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 0, r_next_norm_sq = 0.00188636
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 1, r_next_norm_sq = 0.0025947
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 2, r_next_norm_sq = 0.00193249
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 3, r_next_norm_sq = 0.000252474
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 4, r_next_norm_sq = 0.00011794
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 5, r_next_norm_sq = 0.000202524
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 6, r_next_norm_sq = 4.76587e-05
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 7, r_next_norm_sq = 1.62638e-05
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 8, r_next_norm_sq = 5.91004e-06
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 9, r_next_norm_sq = 2.20483e-06
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 10, r_next_norm_sq = 2.3264e-06
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:526) In linear CG: recomputing residual.
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 11, r_next_norm_sq = 3.41682e-07
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 12, r_next_norm_sq = 1.13422e-07
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 13, r_next_norm_sq = 3.66276e-08
VLOG[5] (VoskAPI:LinearCgd():optimization.cc:528) In linear CG: k = 14, r_next_norm_sq = 6.54284e-08
VLOG[4] (VoskAPI:GetIvector():ivector-extractor.cc:753) Objective function improvement from estimating the iVector (vs. default value) is 14.9481
VLOG[5] (VoskAPI:GetFrame():online-ivector-feature.cc:339) due to --use-most-recent-ivector=true, using iVector from frame 1864 for frame 1863
VLOG[6] (VoskAPI:GetCutoff():lattice-faster-decoder.cc:677) Number of tokens active on frame 42 is 1546
VLOG[6] (VoskAPI:ProcessEmitting():lattice-faster-decoder.cc:729) Adaptive beam on frame 42 is 13
VLOG[6] (VoskAPI:GetCutoff():lattice-faster-decoder.cc:677) Number of tokens active on frame 43 is 1572
VLOG[6] (VoskAPI:ProcessEmitting():lattice-faster-decoder.cc:729) Adaptive beam on frame 43 is 13
ASSERTION_FAILED (VoskAPI:ClearActiveTokens():lattice-faster-decoder.cc:912) Assertion failed: (num_toks_ == 0)
```

**Comment by nshmyrev** [comment]
> To be fair, I have forked the Java code to add a SILENT enum state because I always had warnings when I ended the recognizer

Haha, its pretty cool you reported the issue without mentioning this fact.

Did you try the original library without modification?

> LibVosk.end(model);

We don't have "end" method in libvosk

> I looked at the C code for Kaldi and it seemed that the value -10 was for not showing any log.

Messages are disabled in libvosk, not in the kaldi code. Here:

https://github.com/alphacep/vosk-api/blob/master/src/model.cc#L71

**Comment by hervegirod** [comment]
About the `end `method, Sorry it was the 
`LibVosk.vosk_model_free(model.getPointer());`
I just encapsulated it in my own code, but it is just what I did.

For the kadli code, I looked at the following code at the beginning of the `KaldiLogHandler `function:
`  if (env.severity > GetVerboseLevel())
      return;
`

Which is why at assumed I could disable the logs. But maybe it is not how it works. Or I did something wrong.
